src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -930,13 +930,18 @@ fn preprocess_link(
     ori_link: &MarkdownLink,
     dox: &str,
 ) -> Option<Result<PreprocessingInfo, PreprocessingError>> {
+    // certain link kinds cannot have their path be urls,
+    // so they should not be ignored, no matter how much they look like urls.
+    // e.g. [https://example.com/] is not a link to example.com.
+    let can_be_url = ori_link.kind == LinkType::Inline || ori_link.kind == LinkType::Autolink;
```

Comment:
```suggestion
    let can_be_url = ori_link.kind != LinkType::ShortcutUnknown &&
        ori_link.kind != LinkType::CollapsedUnknown &&
        ori_link.kind != LinkType::ReferenceUnknown;
```

Reply:
ah, it seems like i underestimated the amount of preprocessing that is done before this point.  i was under the impression we were basically working with the raw event stream, before reference resolution, but ofc that's not right.